Distraction Shield/background.js

```diff
@@ -1,102 +1,137 @@
-//Set that holds the urls to be intercepted
-var blockedSites = new BlockedSiteList();
-var interceptDateList = [];
-var localSettings = new UserSettings();
+import {createNewBlockedSite} from './modules/blockedSiteBuilder';
+import BlockedSiteList from './classes/BlockedSiteList';
+import * as interception from './modules/statistics/interception';
+import * as storage from './modules/storage';
+import UserSettings from  './classes/UserSettings'
+import * as constants from'./constants';
+
+/**
+ * The BlockedSiteList used by the background
+ * @type {BlockedSiteList}
+ */
+let blockedSites = new BlockedSiteList();
+/**
+ * The UserSettings used by the background scrript.
+ * @type {UserSettings}
+ */
+let localSettings = new UserSettings();
+
 
 /* --------------- ------ setter for local variables ------ ---------------*/
 
-setLocalSettings = function(newSettings) {
-    var oldState = localSettings.getState();
-    localSettings.copySettings(newSettings);
-    if (oldState != localSettings.getState()) {
+export function setLocalSettings(newSettings) {
+    let oldState = localSettings.state;
+    localSettings = newSettings;
+    if (oldState != localSettings.state) {
+        localSettings.reInitTimer(replaceListener);
         replaceListener();
     }
-};
+}
 
-setLocalBlacklist = function(newList) {
-    blockedSites.setList(newList.getList());
+function setLocalBlacklist(newList) {
+    blockedSites.list = newList.list;
     replaceListener();
-};
-
-setLocalInterceptDateList = function(dateList) {
-    interceptDateList = dateList;
-};
+}
 
 /* --------------- ------ Storage retrieval ------ ---------------*/
-// Methods here are only used upon initialization of the session.
-// Usage found in init.js
 
-retrieveSettings = function(callback, param) {
-    storage.getSettings(function(settingsObject) {
-        localSettings = settingsObject;
-        return callback(param);
-    });
-};
-
-retrieveBlockedSites = function(callback){
-    storage.getBlacklist(function(blacklist) {
-        blockedSites.setList(blacklist.getList());
+export function retrieveBlockedSites(callback) {
+    storage.getBlacklist(function (blacklist) {
+        blockedSites.list = blacklist.list;
         return callback();
     });
-};
-
+}
 /* --------------- ------ Updating of variables ------ ---------------*/
 
-addUrlToBlockedSites = function(unformattedUrl, onSuccess) {
-    blockedSiteBuilder.createNewBlockedSite(unformattedUrl, function(newBS) {
+function addUrlToBlockedSites(unformattedUrl, onSuccess) {
+    createNewBlockedSite(unformattedUrl, function (newBS) {
         if (blockedSites.addToList(newBS)) {
-            synchronizer.syncBlacklist(blockedSites);
+            replaceListener();
```

Comment:
because the listener only needs to be there when the extension is 'on'. Also whenever changes are made to the blockedSiteList, this listener needs to be updated, this can only be done by replacing it with a new one.